pom.xml

```diff
@@ -154,6 +171,19 @@
                 <scala.version>${scala_2.12.version}</scala.version>
             </properties>
         </profile>
+        <profile>
+            <id>spark-3</id>
+            <properties>
+                <spark.compat.version>spark-3</spark.compat.version>
+                <spark.version>${spark_3.version}</spark.version>
+                <spark.sql.mongo.version>${spark_3.sql.mongo.version}</spark.sql.mongo.version>
```

Comment:
when you activate the build with profile `spark-3` you do not deactivate the `spark-2` modules, which means you use spark3 versions of these libraries for the build of spark2 modules as well. Right?

Reply:
Ah, I notice that I didn't test the spark 2.4 / scala 2.12 combination. It won't work in the current configuration.

The idea is that all modules except `compatibility_spark-2`, `compatibility_spark-3`, `hyperdrive-release_spark-2` and `hyperdrive-release_spark-3` are compatible for both spark 2.4 and 3.0. So e.g. if you only depend on the API module, you don't have to worry about spark 2.4 vs 3.0. `compatibility_spark-2` always uses spark 2 dependencies, even if profile `spark-3` is activated, and `hyperdrive-release_spark-2` should always use the `compatibility_spark-2` module. This latest part I forgot to implement, I'll fix this